From a89c1a636c537809bef5676bed71f66388df2ceb Mon Sep 17 00:00:00 2001
From: Danny Nold <dannynold@freescale.com>
Date: Sun, 22 Aug 2010 21:17:35 -0500
Subject: [PATCH] ENGR00126260-1 - MSL change to define E Ink panels in platform code

Shift panel definitions to platform code, and expand panel structure
to encapsulate all variable parameters for configuring the EPDC to drive
the panel.

Signed-off-by: Danny Nold <dannynold@freescale.com>
---
 arch/arm/mach-mx5/mx50_arm2.c |   46 +++++++++++++++++++++++++++++++++++++++++
 1 files changed, 46 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-mx5/mx50_arm2.c b/arch/arm/mach-mx5/mx50_arm2.c
index 844ad5a..0eff14a 100644
--- a/arch/arm/mach-mx5/mx50_arm2.c
+++ b/arch/arm/mach-mx5/mx50_arm2.c
@@ -677,8 +677,54 @@ void epdc_disable_pins(void)
 	gpio_direction_output(EPDC_SDCE2, 0);
 }
 
+static struct fb_videomode e60_mode = {
+	.name = "E60",
+	.refresh = 50,
+	.xres = 800,
+	.yres = 600,
+	.pixclock = 20000000,
+	.left_margin = 10,
+	.right_margin = 217,
+	.upper_margin = 4,
+	.lower_margin = 10,
+	.hsync_len = 20,
+	.vsync_len = 4,
+	.sync = 0,
+	.vmode = FB_VMODE_NONINTERLACED,
+	.flag = 0,
+};
+
+static struct fb_videomode e97_mode = {
+	.name = "E97",
+	.refresh = 50,
+	.xres = 1200,
+	.yres = 825,
+	.pixclock = 32000000,
+	.left_margin = 8,
+	.right_margin = 125,
+	.upper_margin = 4,
+	.lower_margin = 17,
+	.hsync_len = 20,
+	.vsync_len = 4,
+	.sync = 0,
+	.vmode = FB_VMODE_NONINTERLACED,
+	.flag = 0,
+};
+
+static struct mxc_epdc_fb_mode panel_modes[] = {
+	{
+		&e60_mode,
+		4, 10, 20, 10, 20, 480, 20, 0, 1, 1,
+	},
+	{
+		&e97_mode,
+		8, 10, 20, 10, 20, 580, 20, 0, 1, 3,
+	},
+};
 
 static struct mxc_epdc_fb_platform_data epdc_data = {
+	.epdc_mode = panel_modes,
+	.num_modes = ARRAY_SIZE(panel_modes),
 	.get_pins = epdc_get_pins,
 	.put_pins = epdc_put_pins,
 	.enable_pins = epdc_enable_pins,
-- 
1.5.4.4

